code/go/internal/spectypes/filesize.go

```diff
@@ -0,0 +1,121 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package spectypes
+
+import (
+	"encoding/json"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Common units for file sizes.
+const (
+	Byte     = FileSize(1)
+	KiloByte = 1024 * Byte
+	MegaByte = 1024 * KiloByte
+)
+
+const (
+	byteString     = "B"
+	kiloByteString = "KB"
+	megaByteString = "MB"
+)
+
+// FileSize represents the size of a file.
+type FileSize uint64
+
+func parseFileSizeInt(s string) (uint64, error) {
+	// os.FileInfo reports size as int64, don't support bigger numbers.
+	maxBitSize := 63
+	return strconv.ParseUint(s, 10, maxBitSize)
+}
+
+// MarshalJSON implements the json.Marshaler interface for FileSize.
+func (s FileSize) MarshalJSON() ([]byte, error) {
+	return []byte(`"` + s.String() + `"`), nil
+}
+
+// MarshalYAML implements the yaml.Marshaler interface for FileSize.
+func (s FileSize) MarshalYAML() (interface{}, error) {
+	return s.String(), nil
+}
```

Comment:
I admit I'm a bit lost here. Do we need to implement marshallers to check the file size?

Reply:
The marshalers are not really needed here, I added them for completion, they come almost for free after implementing `String()`. `String()` is helpful to show the file sizes in error messages.

Well, thinking twice they may be needed in the places where we convert from JSON to YAML by unmarshaling and marshaling.

Reply:
Maybe add a comment there (why we need marshallers), otherwise every next person will be lost while reading it :)